### PR TITLE
feat: enable use of smtp endpoints for sending mail

### PR DESCRIPTION
**Pull request by radicand**
This code enables use of a SMTP gateway for sending email, avoiding the need for a SaaS email product where not necessary.

### CODE REVIEW

**Comment by jordanh** [comment]
@radicand thank you so much for this! This has been on our minds for a bit. We'll have a look

File: packages/server/email/MailManagerSMTP.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    secure: false,
+    pool: true,
+    maxConnections: 10,
+    rateLimit: 50,
+    tls:
+      process.env.MAIL_SMTP_USE_TLS === '1'
+        ? {
+            rejectUnauthorized: false,
+            ciphers:
+              process.env.MAIL_SMTP_CIPHERS ?? 'HIGH:MEDIUM:!aNULL:!eNULL:@STRENGTH:!DH:!kEDH'
```
Comment (mattkrick) [inline packages/server/email/MailManagerSMTP.ts:24]: could you remove the fallback value here? if the env var is missing we want to know!

**Review by mattkrick — CHANGES_REQUESTED** [review]
thanks for the PR @radicand!
so glad the codebase was easy enough to dig into.
would love to hear more about how you're using parabol!

File: packages/server/email/MailManagerSMTP.ts (hunk 1 above)
Comment (radicand) [reply packages/server/email/MailManagerSMTP.ts]: Good catch, it's removed now.

**Pull request merged**